### DISCUSSION TITLE
Proportional on measurement method

### DISCUSSION THREAD

**Discussion by tominogino** [ScratMan/HASmartThermostat#97] [Ideas]
Hi!
In the description this blog is mentioned:
http://brettbeauregard.com/blog/category/pid/
in which there is this post:
http://brettbeauregard.com/blog/2017/06/introducing-proportional-on-measurement/

The "proportional on measurement" method seems to be interesting and looks promising, as it should overcome the overshoot problem not needing the integral term to go back to zero. On the other hand it's tricky, since you have to define an instant in order to calculate the difference for the proportional term.

I couldn't find it in the current release, am I missing it? If not, are you planning on adding it?

Thank you so much for all your work.
Tom

**Comment by dzerovibe** [comment]
I was about to make the same request, and came across this one.

@ScratMan Do you have any interest in adding this option? It appears to be already part of the Arduino PID Library, on which your code seems to be based. It might be something very simple for you to add. I was thinking of trying it myself, but my coding skills are really not great and very rusty, so I thought I would ask first if you had any interest in looking at it.
